### ISSUE TITLE
Test failure: JIT\\Directed\\StructABI\\structreturn\\structreturn.cmd

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#37880]
failed in job: [runtime-coreclr jitstress 20200613.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=686806&view=ms.vss-test-web.build-test-results-tab&runId=21324670&resultId=106408&paneView=debug)

Error message
~~~
Assert failure(PID 6076 [0x000017bc], Thread: 21484 [0x53ec]): Assertion failed '!"Incompatible types for gtNewTempAssign"' in 'System.Numerics.Vector1[Vector4][System.Numerics.Vector4]:op_Addition(System.Numerics.Vector1[Vector4],System.Numerics.Vector1[Vector4]):System.Numerics.Vector1[Vector4]' during 'Morph - Global' (IL size 3789)

 File: F:\workspace\_work\1\s\src\coreclr\src\jit\gentree.cpp Line: 15164
 Image: C:\dotnetbuild\work\AFB70911\p\CoreRun.exe


Return code: 1
Raw output file: C:\dotnetbuild\work\AFB70911\w\AE7B0975\e\JIT\Directed\Reports\JIT.Directed\StructABI\structreturn\structreturn.output.txt
Raw output:
BEGIN EXECUTION
 "C:\dotnetbuild\work\AFB70911\p\corerun.exe" structreturn.dll 
 [002365] ----G+------ * LCL_FLD simd16 V124 tmp101 [+0]
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\dotnetbuild\work\AFB70911\p
> C:\dotnetbuild\work\AFB70911\w\AE7B0975\e\JIT\Directed\StructABI\structreturn\structreturn.cmd
Expected: True
Actual: False


Stack trace
   at JIT_Directed._StructABI_structreturn_structreturn_._StructABI_structreturn_structreturn_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.arm64.Checked\TestWrappers\JIT.Directed\JIT.Directed.XUnitWrapper.cs:line 55527
~~~

**Comment by sandreenko** [comment]
Probably another case of `4 example` from  https://github.com/dotnet/runtime/issues/37341 but with minopts.

**Comment by v-haren** [comment]
failed again in job: [runtime-coreclr jitstress 20200620.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=697564&view=ms.vss-test-web.build-test-results-tab&runId=21620142&resultId=107698&paneView=debug)

failed tests:
JIT/Directed/StructABI/structreturn/structreturn.sh
JIT\\Directed\\StructABI\\structreturn\\structreturn.cmd


Error message
~~~
Assert failure(PID 3927 [0x00000f57], Thread: 3927 [0x0f57]): Assertion failed '!"Incompatible types for gtNewTempAssign"' in 'System.Numerics.Vector1[Vector4][System.Numerics.Vector4]:op_Addition(System.Numerics.Vector1[Vector4],System.Numerics.Vector1[Vector4]):System.Numerics.Vector1[Vector4]' during 'Morph - Global' (IL size 3789)

 File: /__w/1/s/src/coreclr/src/jit/gentree.cpp Line: 15177
 Image: /root/helix/work/correlation/corerun

apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f7dc0dbd4)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f7dc0dbd4)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f7dc02614)
/root/helix/work/workitem/JIT/Directed/StructABI/structreturn/structreturn.sh: line 317: 3927 Aborted $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code: 1
Raw output file: /root/helix/work/workitem/JIT/Directed/Reports/JIT.Directed/StructABI/structreturn/structreturn.output.txt
Raw output:
BEGIN EXECUTION
/root/helix/work/correlation/corerun structreturn.dll ''
 [002365] ----G+------ * LCL_FLD simd16 V124 tmp101 [+0]
Writing minidump to file /home/helixbot/dotnetbuild/dumps/coredump.3927.dmp
Written 54992896 bytes (13426 pages) to core file
Dump successfully written
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/root/helix/work/correlation
> /root/helix/work/workitem/JIT/Directed/StructABI/structreturn/structreturn.sh
Expected: True
Actual: False


Stack trace
   at JIT_Directed._StructABI_structreturn_structreturn_._StructABI_structreturn_structreturn_sh() in /__w/1/s/artifacts/tests/coreclr/Linux.arm64.Checked/TestWrappers/JIT.Directed/JIT.Directed.XUnitWrapper.cs:line 53781
~~~

**Comment by BruceForstall** [comment]
I also see this in the libraries JIT stress tests,

https://dev.azure.com/dnceng/public/_build/results?buildId=714263&view=ms.vss-test-web.build-test-results-tab&runId=22139664&resultId=172900&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab

Minopts Linux arm64:
```
               [002326] ----G+------              *  LCL_FLD   simd16 V105 tmp82       [+0]

Assert failure(PID 134 [0x00000086], Thread: 148 [0x0094]): Assertion failed '!"Incompatible types for gtNewTempAssign"' in 'System.Numerics.Vector`1[Boolean][System.Boolean]:op_Division(System.Numerics.Vector`1[Boolean],System.Numerics.Vector`1[Boolean]):System.Numerics.Vector`1[Boolean]' during 'Morph - Global' (IL size 3794)

    File: /__w/1/s/src/coreclr/src/jit/gentree.cpp Line: 15176
    Image: /root/helix/work/correlation/dotnet
```

Tiered compilation, JitStress=1, Linux arm64:
```
               [000002] --CXG+------              *  CALL      simd16 System.Numerics.Vector`1[Boolean][System.Boolean].op_ExclusiveOr
     (  3,  4) [000008] ------------ arg0 in d0   +--*  LCL_FLD   simd16 V02 tmp1         [+0]
     (  3,  4) [000001] ----G------- arg1 in d1   \--*  LCL_FLD   simd16 V00 arg0         [+0]

Assert failure(PID 133 [0x00000085], Thread: 148 [0x0094]): Assertion failed '!"Incompatible types for gtNewTempAssign"' in 'System.Numerics.Vector`1[Boolean][System.Boolean]:op_OnesComplement(System.Numerics.Vector`1[Boolean]):System.Numerics.Vector`1[Boolean]' during 'Morph - Global' (IL size 12)

    File: /__w/1/s/src/coreclr/src/jit/gentree.cpp Line: 15176
    Image: /root/helix/work/correlation/dotnet
```

**Comment by BruceForstall** [comment]
@sandreenko Any chance you'll look at this soon? It's affecting a lot of libraries test runs.

**Comment by sandreenko** [comment]
@sandreenko Any chance you'll look at this soon? It's affecting a lot of libraries test runs.

It will be closed with #37745 that I believe will be merged ~tomorrow.
